hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -406,59 +402,65 @@ protected void advance(long jobId, int count) {
         instance.get().getMap(ADVANCE_MAP_NAME).put(jobId, count);
     }
 
-    protected static BatchSource<Integer> batchSource() {
+    protected static DAG batchSource() {
         return batchSource(r -> { });
     }
 
-    protected static BatchSource<Integer> batchSource(ConsumerEx<Integer> action) {
-        return SourceBuilder.batch("batchSource",
-                        ctx -> new TestSource(ctx, action, false))
-                .fillBufferFn(TestSource::fillBuffer)
-                .build();
+    protected static DAG batchSource(ConsumerEx<Integer> action) {
+        return new DAG().vertex(new Vertex("batchSource",
+                forceTotalParallelismOne(new TestProcessorSupplier(action, false))));
     }
 
-    protected static BatchSource<Integer> streamSource() {
+    protected static DAG streamSource() {
         return streamSource(r -> { });
     }
 
-    protected static BatchSource<Integer> streamSource(ConsumerEx<Integer> action) {
-        return SourceBuilder.batch("streamSource",
-                        ctx -> new TestSource(ctx, action, true))
-                .fillBufferFn(TestSource::fillBuffer)
-                .build();
+    protected static DAG streamSource(ConsumerEx<Integer> action) {
+        return new DAG().vertex(new Vertex("streamSource",
+                forceTotalParallelismOne(new TestProcessorSupplier(action, true))));
     }
 
-    static class TestSource {
-        final Runnable incrementRunCount;
-        final Supplier<Boolean> advance;
+    static class TestProcessorSupplier implements ProcessorSupplier {
```

Comment:
Maybe it would be better to refactor this test to use MockP. It would simplify the test quite a bit.